src/plugins/safeSnap/components/SafeTransactions.vue

```diff
@@ -69,21 +69,73 @@ export const ensureRightNetwork = async chainId => {
   }
 };
 
-async function fetchBalances(network, gnosisSafeAddress) {
-  if (gnosisSafeAddress) {
-    try {
-      const balances = await getGnosisSafeBalances(network, gnosisSafeAddress);
-      return balances
-        .filter(balance => balance.token)
-        .map(balance => ({
-          ...balance.token,
-          address: balance.tokenAddress
-        }));
-    } catch (e) {
-      console.warn('Error fetching balances');
-    }
+async function fetchBalances(network, safeAddress) {
+  if (!safeAddress) {
+    return [];
+  }
+
+  try {
+    const balances = await getGnosisSafeBalances(network, safeAddress);
+
+    const uniswapTokensPromise = fetchTokens(
+      'https://gateway.ipfs.io/ipns/tokens.uniswap.org'
+    );
+    const snapshotTokensPromise = fetchTokens(
+      'https://sh5.co/api/moderation?list=verifiedTokens'
+    );
+
+    const tokensLists = await Promise.all([
+      uniswapTokensPromise,
+      snapshotTokensPromise
+    ]);
+    const tokens = tokensLists.flat();
+
+    return sortBalances(enhanceBalances(balances, tokens));
+  } catch (e) {
+    console.warn('Error fetching balances');
+    return [];
   }
-  return [];
+}
+
+function fetchTokens(url, isNested) {
+  return fetch(url)
+    .then(response => response.json())
+    .then(data => {
+      if (isNested && 'verifiedTokens' in data) {
+        return data.verifiedTokens.tokens;
+      }
+      return 'tokens' in data ? data.tokens : [];
+    })
+    .catch(() => []);
+}
+
+function enhanceBalances(balances, tokens) {
+  return balances
+    .filter(balance => balance.token)
+    .map(balance => enhanceBalance(balance, tokens));
+}
+
+function enhanceBalance(balance, tokens) {
+  const verifiedToken = getVerifiedToken(balance.tokenAddress, tokens);
+  return {
+    ...balance.token,
+    address: balance.tokenAddress,
+    balance: balance.balance
+      ? formatUnits(balance.balance, balance.token.decimals)
```

Comment:
When displaying the token balances in the modal list there seems to be a rounding to only 2 decimals.

I suppose this is not coming from the `formatUnits` function above, but more from the TreasuryAssetsListItem.vue component, but I'm unsure:
https://github.com/snapshot-labs/snapshot/pull/3878/files#diff-7cee60ddbcbb66ec353d818b8b0343f48ad9729d9e8c99ec705a6586f7419bf7R29

This is not very problematic, except that it can become a bit misleading with tokens like WBTC
<img width="436" alt="image" src="https://github.com/snapshot-labs/snapshot/assets/20560366/2a291dfc-0c52-4fa6-ada7-f5dbe6656daa">
Or even worse for tokens with small balances that will be rounded to zero:
<img width="477" alt="image" src="https://github.com/snapshot-labs/snapshot/assets/20560366/cd582b7d-7d57-4a79-ab12-49b586598d66">

You can see that the actual SAFE is showing balances with adaptative decimals so that it gives an accurate picture of the balances in question:
<img width="1361" alt="image" src="https://github.com/snapshot-labs/snapshot/assets/20560366/3d40de93-27fe-45c2-8527-84dd22dc45e3">
https://app.safe.global/balances?safe=eth:0x0DA0C3e52C977Ed3cBc641fF02DD271c3ED55aFe

Reply:
Yep good point. I've made it 6 decimals now